pyscraper/committees/commons_api.py

```diff
@@ -0,0 +1,313 @@
+    scrutinising_departments: List[Department] = []
+    is_lead_committee: Optional[bool] = None
+    name_history: List[NameHistory]
+    members: List[Member] = Field(default_factory=list)
+
+    def parl_url(self):
+        return f"https://committees.parliament.uk/committee/{self.id}/"
+
+    def expand(self):
+        """
+        Get the full committee information from the API
+        """
+        url = f"https://committees-api.parliament.uk/api/Committees/{self.id}?includeBanners=false&showOnWebsiteOnly=false"
+        response = httpx.get(url, timeout=30)
+        response.raise_for_status()
+        data = response.json()
+        return Committee.model_validate(data).add_members()
+
+    def add_members(self):
+        """
+        Fetch and add members to the committee.
+        """
+
+        def one_request(skip=0):
+            url = f"https://committees-api.parliament.uk/api/Committees/{self.id}/Members?MembershipStatus=Current&ShowOnWebsiteOnly=true&skip={skip}"
+            response = httpx.get(url, timeout=30)
+            response.raise_for_status()
+            return response.json()
+
+        data = committee_app_loop(one_request)
+        self.members = [Member.model_validate(item) for item in data]
+        return self
+
+
+class CommitteeList(RootModel[list[Committee]]):
+    def expand(self):
+        new_items = []
+
+        for item in tqdm(self.root, disable=not VerboseSettings.verbose):
+            expanded = item.expand()
+            new_items.append(expanded)
+
+        return CommitteeList(root=new_items)
+
+    def __iter__(self):
+        return iter(self.root)
+
+
+class ApiCallWithSkip(Protocol):
+    def __call__(self, skip: int) -> dict[Any, Any]: ...
+
+
+def committee_app_loop(func: ApiCallWithSkip):
+    items: list[dict[str, str]] = []
+    skip = 0
+    pbar = tqdm(total=None, leave=False, disable=not VerboseSettings.verbose)
+    while True:
+        batch = func(skip=skip)
+        if not batch["items"]:
+            break
+        pbar.total = batch["totalResults"]
+        items.extend(batch["items"])
+        skip += len(batch["items"])
+        pbar.update(len(batch["items"]))
+    pbar.close()
+    return items
+
+
+def get_committee_all_items():
+    def get_committees(skip=0):
+        url = f"https://committees-api.parliament.uk/api/Committees?CommitteeStatus=Current&skip={skip}"
+        response = httpx.get(url)
+        response.raise_for_status()
+        return response.json()
+
+    committees = committee_app_loop(get_committees)
+
+    committee_json_path.parent.mkdir(parents=True, exist_ok=True)
+
+    committee_list = CommitteeList.model_validate(committees)
+
+    committee_list = committee_list.expand()
+
+    with open(committee_json_path, "w") as f:
+        f.write(committee_list.model_dump_json(indent=2))
+
+
+def convert_to_groups():
+    """
+    Convert to the reduced groups format
+    """
+    with committee_json_path.open("r") as f:
```

Comment:
Are we doing this to streamline processing later on and remove the information we don't really care about?

Reply:
Yes - It's trying to get down to the minimum we'd need to display a 'group [x] and their membership' page as a format that would work for other parliaments or groups like APPGs where we can't be sure about membership dates, and there may be some extra-parliamentary members (which also applies to some committees here) - so the current ministers-2010.json can't hold all the data we need.